src/Runner.Worker/JobRunner.cs

```diff
@@ -84,7 +84,14 @@ public async Task<TaskResult> RunAsync(AgentJobRequestMessage message, Cancellat
                 Trace.Info($"Creating job server with URL: {jobServerUrl}");
                 // jobServerQueue is the throttling reporter.
                 _jobServerQueue = HostContext.GetService<IJobServerQueue>();
-                VssConnection jobConnection = VssUtil.CreateConnection(jobServerUrl, jobServerCredential, new DelegatingHandler[] { new ThrottlingReportHandler(_jobServerQueue) });
+                var delegatingHandlers = new List<DelegatingHandler>() { new ThrottlingReportHandler(_jobServerQueue) };
+                message.Variables.TryGetValue("system.github.enable_http_redirects", out VariableValue enableHttpRedirects);
+                if (StringUtil.ConvertToBoolean(enableHttpRedirects?.Value) &&
```

Comment:
is it possible to only enable this for the actions resolve endpoint? or we plan to have the redirect for all requests to Actions service at some point?

Reply:
It would look "hacky" if we try to scope it to a single endpoint. Besides, Actions service controls whether it redirects request or not so what does it buy for us if we add additional constraints on the client side? `VssConnection` is used for accessing Actions service only so it's not like we install system-wide redirection handler that would also engage when talking to other services.